packages/data-context/src/sources/RelevantRunsDataSource.ts

```diff
@@ -180,31 +181,41 @@ export class RelevantRunsDataSource {
   /**
    * Clear the cached current run to allow the data source to pick the next completed run as the current
    */
-  //TODO figure out how to mock in test so do not need to pass in values here
-  async moveToNext (shasForTest?: string[]) {
+  async moveToNext (shas: string[]) {
     debug('Moving to next relevant run')
-    this.#currentRun = undefined
-    this.#currentCommitSha = undefined
 
-    const runs = await this.getRelevantRuns(this.ctx.git?.currentHashes || shasForTest || [])
+    await this.checkRelevantRuns(shas, true)
+  }
+
+  async checkRelevantRuns (shas: string[], clearCache: boolean = false) {
+    debug(`check relevant runs with ${shas.length} shas and clear cache set to ${clearCache}`)
+    if (clearCache) {
+      this.#currentRun = undefined
+      this.#currentCommitSha = undefined
+    }
+
+    const runs = await this.getRelevantRuns(shas)
+
+    //TODO handle error from remote call
```

Comment:
Is there anything left for this TODO? If yes, we could make an issue and link it here.

Reply:
It is handled in the method itself now, so removing this TODO